app/lib/fetcher.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'json'
+require 'tmpdir'
+require 'openssl'
+
+OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE
+
+##
+# Fetch research data objects from DataSpace
+class Fetcher
+  REST_LIMIT = 100
+  attr_reader :server
+
+  # @param [Hash] opts  options to pass to the client
+  # @option opts [String] :server ('https://dataspace.princeton.edu/rest/')
+  def initialize(server:)
+    @server = server
+  end
+
+  # @param id [String] collection id
+  # @return [Array<Hash>] metadata hash for each record
+  def fetch_collection(id)
```

Comment:
Question: How does fetching collections from the REST service relate to OAI endpoints? Just a point of curiosity and probably something that will become clearer later.

Reply:
Ah, it doesn't. Again, good point. I am trying to copy what Jon did in orangetheses, and sort of feeling my way forward. You are correct, we are not using OAI here and I've removed the references to it.